Tools/Algorithms/parser.py

```diff
@@ -0,0 +1,31 @@
+def parse_input(input_file):
```

Comment:
I would make the file name more descriptive, something like `input_file_parser.py`. And the function name `parse_input_file`.